subprojects/core/src/test/groovy/org/gradle/api/internal/tasks/options/OptionReaderTest.groovy

```diff
@@ -169,6 +185,15 @@ class OptionReaderTest extends Specification {
         e.message == "Option 'stringValue' on method cannot take multiple parameters in class 'org.gradle.api.internal.tasks.options.OptionReaderTest\$TestClass4#setStrings'."
     }
 
+    def "fails when option name starts with 'no-'"() {
```

Comment:
:x: This test should be removed now. A new one that assert that we don't add the disabling option dynamically if there's already an option with that name should be added.

Reply:
The new test mentioned here is still missing. Or did I miss it?

Reply:
That's added now, I left it for after the implementation changes :)